### ISSUE TITLE
llvm-mc not found errors when gathering info

### ISSUE THREAD

**Issue by mattgodbolt** [mattgodbolt/compiler-explorer#762]
```
Jan 09 07:22:24 bcbcca9b180f unified:  error: Unable to get version for compiler '/opt/compiler-explorer/clang+llvm-3.0-x86_64-linux-Ubuntu-11_10/bin/llvm-mc' - non-zero result 1 
Jan 09 07:22:24 bcbcca9b180f unified:  info: Gathering version information on /opt/compiler-explorer/clang+llvm-3.4.1-x86_64-unknown-ubuntu12.04/bin/llvm-mc 
Jan 09 07:22:24 bcbcca9b180f unified:  error: Unable to get version for compiler '/opt/compiler-explorer/clang-3.2/bin/llvm-mc' - non-zero result 1 
Jan 09 07:22:24 bcbcca9b180f unified:  info: Gathering version information on /opt/compiler-explorer/clang+llvm-3.5.1-x86_64-linux-gnu/bin/llvm-mc 
Jan 09 07:22:24 bcbcca9b180f unified:  error: Unable to get version for compiler '/opt/compiler-explorer/clang+llvm-3.1-x86_64-linux-ubuntu_12.04/bin/llvm-mc' - non-zero result 1 
Jan 09 07:22:24 bcbcca9b180f unified:  info: Gathering version information on /opt/compiler-explorer/clang+llvm-3.5.0-x86_64-linux-gnu/bin/llvm-mc 
Jan 09 07:22:24 bcbcca9b180f unified:  error: Unable to get version for compiler '/opt/compiler-explorer/clang-3.3/bin/llvm-mc' - non-zero result 1 
```

**Comment by partouf** [comment]
Just checked the download for clang 3.0, and the bin folder does contain llvm-mc, maybe execution with --version returns an error in the older versions? Would be strange if so.

**Comment by mattgodbolt** [comment]
```
ubuntu@admin-node ~> /opt/compiler-explorer/clang+llvm-3.0-x86_64-linux-Ubuntu-11_10/bin/llvm-mc --version
Low Level Virtual Machine (http://llvm.org/):
  llvm version 3.0
  Optimized build.
  Host: x86_64-unknown-linux-gnu
  Host CPU: i686
ubuntu@admin-node ~> echo $status
1
```

Seems like running `llvm-mc` returns an error code (the above here is the fish shell equivalent of `echo $?`)

**Comment by partouf** [comment]
That seems problematic... :/ what to do about that

**Comment by RabsRincon** [comment]
So two problems here: `llvm-mc --version` from clang 3.0 returned `llvm` in lowercase, which threw the regex out. That's now fixed by making the version parsing case insensitive. As for the 1 code return, I've changed the flow of the initialision so a non-0 code will still check for the version flag. If the error code was a real error code and no version is reported, no harm done, as it will get filtered out when we check if we indeed have a version.
